server/iotdb/conf/iotdb-engine.properties

```diff
@@ -155,7 +155,43 @@ concurrent_flush_thread=0
 
 # whether take over the memory management by IoTDB rather than JVM when serializing memtable as bytes in memory
 # (i.e., whether use ChunkBufferPool), value true, false
-chunk_buffer_pool_enable = false
+chunk_buffer_pool_enable=false
+
+####################
+### Merge Configurations
+####################
+
+# How many thread will be set up to perform merges, 1 by default.
+# Set to 1 when less than or equal to 0.
+merge_thread_num=1
+
+# How much memory may be used in ONE merge task (in byte), 20% of maximum JVM memory by default.
+# This is only a rough estimation, starting from a relatively small value to avoid OOM.
+# Each new merge thread may take such memory, so merge_thread_num * merge_memory_budget is the
+# total memory estimation of merge.
+# merge_memory_budget=2147483648
+
+# When set to true, if some crashed merges are detected during system rebooting, such merges will
+# be continued, otherwise, the unfinished parts of such merges will not be continued while the
+# finished parts still remains as they are.
+# If you are feeling the rebooting is too slow, set this to false, false by default
+continue_merge_after_reboot=false
+
+# A global merge will be performed each such interval, that is, each storage group will be merged
+# (if proper merge candidates can be found). Unit: second, default: 1hours.
+# When less than or equal to 0, timed merge is disabled.
+merge_interval_sec=3600
+
+# When set to true, all merges becomes full merge (the whole SeqFiles are re-written despite how
```

Comment:
No overflow, no merge. The most and certain merit of doing merge is to make the unsequential files sequential.
Making the deleted data disappear physically is fine, but it would be considered as a future work.